From 8fb451af770502638f81d99ae76a50198f0de326 Mon Sep 17 00:00:00 2001
From: Liang Chen <cl@rock-chips.com>
Date: Fri, 24 Nov 2017 11:37:02 +0800
Subject: [PATCH] ARM: dts: rk312x: adjust leakage-scaling-sel and
 leakage-voltage-sel

Adjust leakage-scaling-sel and leakage-voltage-sel for the chips with
low leakage(<14mA).

Change-Id: Id73691e0e13d6b37916c594636f43d22762d9ad1
Signed-off-by: Liang Chen <cl@rock-chips.com>
---
 arch/arm/boot/dts/rk312x.dtsi | 43 +++++++++++++++++++++++------------
 1 file changed, 28 insertions(+), 15 deletions(-)

diff --git a/arch/arm/boot/dts/rk312x.dtsi b/arch/arm/boot/dts/rk312x.dtsi
index 69735f0a161e..6a8cdd6189f2 100644
--- a/arch/arm/boot/dts/rk312x.dtsi
+++ b/arch/arm/boot/dts/rk312x.dtsi
@@ -97,9 +97,15 @@
 		compatible = "operating-points-v2";
 		opp-shared;
 
+		rockchip,leakage-scaling-sel = <
+			1   13   18
+			14  254  0
+		>;
+		clocks = <&cru PLL_APLL>;
 		rockchip,leakage-voltage-sel = <
-			1   18   0
-			18  254  1
+			1   13   0
+			14  18   1
+			18  254  2
 		>;
 		nvmem-cells = <&cpu_leakage>;
 		nvmem-cell-names = "cpu_leakage";
@@ -109,6 +115,7 @@
 			opp-microvolt = <950000>;
 			opp-microvolt-L0 = <950000>;
 			opp-microvolt-L1 = <950000>;
+			opp-microvolt-L2 = <950000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-408000000 {
@@ -116,42 +123,48 @@
 			opp-microvolt = <950000>;
 			opp-microvolt-L0 = <950000>;
 			opp-microvolt-L1 = <950000>;
+			opp-microvolt-L2 = <950000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-600000000 {
 			opp-hz = /bits/ 64 <600000000>;
-			opp-microvolt = <1025000>;
-			opp-microvolt-L0 = <1025000>;
-			opp-microvolt-L1 = <975000>;
+			opp-microvolt = <1075000>;
+			opp-microvolt-L0 = <1075000>;
+			opp-microvolt-L1 = <1025000>;
+			opp-microvolt-L2 = <975000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-696000000 {
 			opp-hz = /bits/ 64 <696000000>;
-			opp-microvolt = <1100000>;
-			opp-microvolt-L0 = <1100000>;
-			opp-microvolt-L1 = <1050000>;
+			opp-microvolt = <1150000>;
+			opp-microvolt-L0 = <1150000>;
+			opp-microvolt-L1 = <1100000>;
+			opp-microvolt-L2 = <1050000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-816000000 {
 			opp-hz = /bits/ 64 <816000000>;
-			opp-microvolt = <1150000>;
-			opp-microvolt-L0 = <1150000>;
-			opp-microvolt-L1 = <1100000>;
+			opp-microvolt = <1200000>;
+			opp-microvolt-L0 = <1200000>;
+			opp-microvolt-L1 = <1150000>;
+			opp-microvolt-L2 = <1100000>;
 			clock-latency-ns = <40000>;
 			opp-suspend;
 		};
 		opp-1008000000 {
 			opp-hz = /bits/ 64 <1008000000>;
-			opp-microvolt = <1275000>;
-			opp-microvolt-L0 = <1275000>;
-			opp-microvolt-L1 = <1225000>;
+			opp-microvolt = <1325000>;
+			opp-microvolt-L0 = <1325000>;
+			opp-microvolt-L1 = <1275000>;
+			opp-microvolt-L2 = <1225000>;
 			clock-latency-ns = <40000>;
 		};
 		opp-1200000000 {
 			opp-hz = /bits/ 64 <1200000000>;
 			opp-microvolt = <1425000>;
 			opp-microvolt-L0 = <1425000>;
-			opp-microvolt-L1 = <1375000>;
+			opp-microvolt-L1 = <1425000>;
+			opp-microvolt-L2 = <1375000>;
 			clock-latency-ns = <40000>;
 		};
 	};
-- 
2.35.3

